### DISCUSSION TITLE
bindparams usage

### DISCUSSION THREAD

**Discussion by ohadle** [sqlalchemy/sqlalchemy#6686] [Usage Questions] [answered]
Every parameter I set with bindparams seems to be `None`:
```
from sqlalchemy import text
a = text("select :b")
a.bindparams(b='some')
a._bindparams
```
Returns:
`{'b': BindParameter('b', None, type_=NullType())}`

**Comment by CaselIT** [comment, marked as answer]
hi,

sqlalchemy uses a generative pattern, so calling `bindparams` returns a new object with the applied changes.
Try:
```py
from sqlalchemy import text
a = text("select :b")
a = a.bindparams(b='some')
a._bindparams
```

**Reply by ohadle** [reply]
Thanks!
